internal/hcp/registry/types.iterations.go

```diff
@@ -35,40 +38,37 @@ func NewIteration() *Iteration {
 }
 
 // Initialize prepares the iteration to be used with an active HCP Packer registry bucket.
-func (i *Iteration) Initialize(opts IterationOptions) error {
+func (i *Iteration) Initialize() error {
 	if i == nil {
 		return errors.New("Unexpected call to initialize for a nil Iteration")
 	}
 
 	// By default we try to load a Fingerprint from the environment variable.
-	// If no variable is defined we should try to load a fingerprint from Git, or other VCS.
+	// If no variable is defined we generate a new fingerprint.
 	i.Fingerprint = os.Getenv(env.HCPPackerBuildFingerprint)
 
 	if i.Fingerprint != "" {
 		return nil
 	}
 
-	fp, err := GetGitFingerprint(opts)
+	fp, err := ulid.New(ulid.Now(), ulid.Monotonic(rand.New(rand.NewSource(time.Now().UnixNano())), 0))
 	if err != nil {
-		return err
+		return fmt.Errorf("Failed to generate a fingerprint: %s", err)
 	}
-	i.Fingerprint = fp
+	i.Fingerprint = fp.String()
 
 	return nil
 }
 
-// GetGitFingerprint returns the HEAD commit for some template dir defined in opt.TemplateBaseDir.
+// GetGitSHA returns the HEAD commit for some template dir defined in opt.TemplateBaseDir.
 // If the base directory is not under version control an error is returned.
-func GetGitFingerprint(opts IterationOptions) (string, error) {
-	r, err := git.PlainOpenWithOptions(opts.TemplateBaseDir, &git.PlainOpenOptions{
+func GetGitSHA(baseDir string) (string, error) {
```

Comment:
Does it make sense to move this method somewhere else now? Keeping in `types.iterations.go` doesn't seem to make sense anymore.